### DISCUSSION TITLE
Request for Transparency and Management Options for Audio File Retention

### DISCUSSION THREAD

**Discussion by Keithsc** [open-webui/open-webui#6705] [Ideas]
I am currently using the Open-WebUI project in a professional setting, and I have some concerns regarding the retention of audio files generated during the transcription process. Given the sensitivity of audio data, I feel it is crucial to have clarity on how this data is managed. 

Issue Summary:  
Audio File Retention:  I would like to understand how long audio files are retained after the transcription process. Specifically, I’ve noticed that once audio is recorded and transcribed, the files appear to be stored within the following directory: ./data/cache/audio/transcriptions
     
However, it's unclear how long these files remain in this location and whether they accumulate with each session or chat. 

User Control Options: 
    To ensure compliance with privacy standards and provide users with greater control, I propose adding an option within the settings that allows users to specify the duration for which audio files are retained. For example, a user-selectable option could provide choices such as: 
        Delete audio files immediately
        Delete audio files when a chat is deleted
        Delete audio files after x hours / days ?
        Keep audio files indefinitely until manually deleted by the user.
         
Additional Suggestions:  
        It would be beneficial to include a prompt or notification when audio files are being deleted, letting the user know that their files were removed.

        A clear documentation or section within the interface describing how the audio files are handled would provide added transparency to users.
         
In an era where data privacy is paramount, I believe these enhancements would not only bolster user trust but also ensure that our usage of Open-WebUI aligns with best practices in data management. 

Thanks for reading.

**Comment by Classic298** [comment]
Well not only audio files, documents and other files and chats data when deleting the chat would also be nice

**Comment by ips972** [comment]
It would be great if open webui would have a global retention policy page. Divided into sections for chats, images, audio gen, audio recordings. For each the ability to select delete instantly, delete in X days, delete on logoff. 
My chat database is growing at an alarming rate with more than 200 users and running with postgresSQL backend and pgvector. For now its all great but, chat history isn’t summarized for PER user knowledge on the fly.. so why have more than 2 to 3/4 months of history.. if a user wishes to keep a specific chat. Use archive button. 

Hope this gets implemented! Good luck.

**Comment by surak** [comment]
I would like to have the option of not keeping anything from the user, to be honest - like a read-only system. 

It makes no sense to store conversations server-side. It should really stay on the cookies. Saves a lot of headaches for the sysadmins and keeps user data private in case of an invasion.

It's what's keeping me from deploying Open-webui system-wide here

**Comment by flefevre** [comment]
Sorry I didn't see your post. I will close mine
Enhancement: Add configurable retention and storage location for meeting audio notes #14129

**Comment by Classic298** [comment]
Hello everyone.

I worked on a PR related to this discussion.
Testing wanted!

https://github.com/open-webui/open-webui/pull/16520
